File: update_headers.py
import os
import glob
import re

def update_headers():
    # Read index.html to get the new header
    with open('index.html', 'r', encoding='utf-8') as f:
        index_content = f.read()

    # Extract the header block from index.html
    header_pattern = re.compile(r'([ \t]*<!-- 1\. TOP NAVBAR -->.*?</header>)', re.DOTALL)
    match = header_pattern.search(index_content)
    
    if not match:
        print("Could not find header in index.html")
        return
        
    new_header = match.group(1)
    print("Found new header block of length:", len(new_header))

    # Update all other html files
    html_files = glob.glob('*.html')
    html_files.remove('index.html')

    for file_path in html_files:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
            
        # Check if the file has a header block
        if re.search(header_pattern, content):
            # Replace the old header with the new header
            new_content = re.sub(header_pattern, new_header, content)
            
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f"Updated header in {file_path}")
        else:
            print(f"Could not find header block in {file_path}")

if __name__ == '__main__':
    update_headers()

File: update_headers.js
const fs = require('fs');
const path = require('path');

const indexHtml = fs.readFileSync('index.html', 'utf8');
const headerRegex = /[ \t]*<!-- 1\. TOP NAVBAR -->[\s\S]*?<\/header>/;
const match = indexHtml.match(headerRegex);

if (!match) {
    console.error("Could not find header in index.html");
    process.exit(1);
}

const headerContent = match[0];
console.log("Found header length:", headerContent.length);

const files = fs.readdirSync('.').filter(f => f.endsWith('.html') && f !== 'index.html');

for (const file of files) {
    let content = fs.readFileSync(file, 'utf8');
    if (headerRegex.test(content)) {
        content = content.replace(headerRegex, headerContent);
        fs.writeFileSync(file, content, 'utf8');
        console.log("Updated header in " + file);
    } else {
        console.log("Skipped " + file + " - no header found");
    }
}
